internal/service/advancedclustertpf/resource.go

```diff
@@ -54,40 +69,27 @@ func (r *rs) Schema(ctx context.Context, req resource.SchemaRequest, resp *resou
 
 func (r *rs) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
 	var plan TFModel
-	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
-	if resp.Diagnostics.HasError() {
-		return
-	}
-	sdkReq := NewAtlasReq(ctx, &plan, &resp.Diagnostics)
-	if resp.Diagnostics.HasError() {
-		return
-	}
-	err := StoreCreatePayload(sdkReq)
-	if err != nil {
-		resp.Diagnostics.AddError("errorCreate", fmt.Sprintf(errorCreate, err.Error()))
+	diags := &resp.Diagnostics
+	diags.Append(req.Plan.Get(ctx, &plan)...)
+	if diags.HasError() {
 		return
 	}
-	tfNewModel, shouldReturn := mockedSDK(ctx, &resp.Diagnostics, plan.Timeouts)
-	if shouldReturn {
-		return
+	model := r.createCluster(ctx, &plan, diags)
```

Comment:
very nit, but why not have the create logic in the root create function? Applies for read/update as well

Reply:
Trying to keep a consistent higher abstraction in CRUD methods to enhance re-use and make navigation simpler.
(Quite subjective, not sured it has the desired effect 😅)